tests/test_iqm_naive_move_pass.py

```diff
@@ -1,85 +1,403 @@
 """Testing IQM transpilation.
 """
 
-from qiskit.circuit import QuantumCircuit
+from itertools import product
+
+import numpy as np
+import pytest
+import qiskit
+from qiskit.circuit import ClassicalRegister, QuantumCircuit, QuantumRegister
 from qiskit.circuit.library import QuantumVolume
-from qiskit.circuit.quantumcircuitdata import CircuitInstruction
+from qiskit.compiler import transpile
+import qiskit.passmanager
 from qiskit.quantum_info import Operator
+import qiskit.scheduler
+import stevedore
 
-from iqm.qiskit_iqm.iqm_naive_move_pass import transpile_to_IQM
-from iqm.qiskit_iqm.iqm_provider import IQMBackend
+from iqm.iqm_client import ExistingMoveHandlingOptions
+from iqm.qiskit_iqm.iqm_circuit_validation import validate_circuit
+from iqm.qiskit_iqm.iqm_naive_move_pass import _get_scheduling_method, transpile_to_IQM
+from iqm.qiskit_iqm.iqm_transpilation import IQMReplaceGateWithUnitaryPass
+from iqm.qiskit_iqm.move_gate import MOVE_GATE_UNITARY, MoveGate
 
-from .utils import _get_allowed_ops, _is_valid_instruction, get_mocked_backend
+from .utils import capture_submitted_circuits, get_mocked_backend
 
 
-def test_transpile_to_IQM_star_semantically_preserving(ndonis_architecture):  # pylint: disable=too-many-locals
-    backend, _client = get_mocked_backend(ndonis_architecture)
-    qubit_registers = _get_qubit_registers(backend)
-    n_qubits = len(qubit_registers)
-    for i in range(2, n_qubits + 1):
-        circuit = QuantumVolume(i, i)
-        # Use optimization_level=0 to avoid that the qubits get remapped.
-        transpiled_circuit = transpile_to_IQM(circuit, backend, optimization_level=0)
-        transpiled_operator = Operator(transpiled_circuit)
+def get_test_circuit(kind, size):
+    """Generates a test circuit according to the kind and size."""
+    if kind == "QuantumVolume":
+        return QuantumVolume(size)
+    if kind == "GHZ":
+        qc = QuantumCircuit(size)
+        qc.h(0)
+        for i in range(size - 1):
+            qc.cx(i, i + 1)
+        return qc
+    if kind == "MCM":
+        qc = QuantumCircuit(size, 2 * size)
+        qc.h(0)
+        for i in range(1, size):
+            qc.cx(0, i)
+            qc.measure(i, i)
+        for i in range(size):
+            qc.h(i)
+            qc.measure(i, size + i)
+            # Not using measure_all here because that was broken (SW-389)
+        return qc
+    raise ValueError(f"Unknown circuit kind: {kind}")
+
+
+@pytest.fixture()
+def backend(request):
+    """Fixture that returns a mocked backend."""
+    return get_mocked_backend(request.getfixturevalue(request.param))[0]
+
+
+@pytest.mark.parametrize(
+    ("circuit_kind", "circuit_size", "backend", "transpile_method"),
+    list(
+        product(
+            ["GHZ", "QuantumVolume", "MCM"],
+            [2, None],  # Try a small circuit and the largest possible circuit
```

Comment:
```suggestion
            [2, 5],  
```

Some of these tests simulate the states, so the max size should be fixed.